### PR TITLE
Steplist css improvements / HCM

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Addresses some of the follow up items we noted.
- HCM
- label wrapping (different than steplist wrapping)
- marker alignment
- dashes should have rounded ends
- colors in index file, also non-colors in skin file
- simplified some of the property assignments to make t-shirt sizes easier to understand
- added comments
- do a little less overriding between vertical/horizontal, normal/emphasized, and selected/disabled/etc

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by dannify — APPROVED** [review]
Looks good on WHCM

**Review by LFDanLu — COMMENTED** [review]
looks good to me, tested with and w/o HCM , just one comment

File: packages/@react-spectrum/steplist/stories/StepList.stories.tsx [hunk 1]
```diff
@@ -96,21 +96,72 @@ export const DefaultCompleted: DefaultStory = {
...
+export const HorizontalWithOddLengths: StepListStory = {
+  args: {
+    width: 600,
+    children: options.map((o) => <Item key={o.key}>{`${o.value} ${o.value} ${o.value}`}</Item>),
+    defaultSelectedKey: options[1].key,
+    defaultLastCompletedStep: options[1].key
+  },
+  name: 'Horizontal Odd names'
+};
+
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/steplist/stories/StepList.stories.tsx:164]: good question, i did purposely punt on it for this PR and haven't had a chance to bring it up with design yet.
thanks for the reminder though, I need to add that to the agenda

**Review by LFDanLu — APPROVED** [review]

File: packages/@react-spectrum/steplist/stories/StepList.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/steplist/stories/StepList.stories.tsx:164]: sounds good

**Pull request merged**